pkg/client/client.go

```diff
@@ -37,66 +36,139 @@ import (
+	providerForNutanixCluster, err := n.buildProviderFromNutanixCluster(nutanixCluster)
+	if err != nil {
+		return nil, fmt.Errorf("error building an environment provider from NutanixCluster: %w", err)
+	}
+	if providerForNutanixCluster != nil {
+		providers = append(providers, providerForNutanixCluster)
 	} else {
 		log.Info(fmt.Sprintf("[WARNING] prismCentral attribute was not set on NutanixCluster %s in namespace %s. Defaulting to CAPX manager credentials", nutanixCluster.Name, nutanixCluster.Namespace))
 	}
 
-	// Add env provider for CAPX manager
-	npe, err := n.getManagerNutanixPrismEndpoint()
+	// Fallback to building a provider using the global CAPX manager credentials
+	providerForLocalFile, err := n.buildProviderFromFile()
+	if err != nil {
+		return nil, fmt.Errorf("error building an environment provider from file: %w", err)
+	}
+	if providerForLocalFile != nil {
+		providers = append(providers, providerForLocalFile)
+	}
+
+	// Initialize environment with providers
+	env := environment.NewEnvironment(providers...)
+	// GetManagementEndpoint will return the first valid endpoint from the list of providers
+	me, err := env.GetManagementEndpoint(envTypes.Topology{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get management endpoint object: %w", err)
+	}
+	return me, nil
+}
+
+// buildProviderFromNutanixCluster will return an envTypes.Provider with info from the provided NutanixCluster.
+// It will return nil if nutanixCluster.Spec.PrismCentral is nil.
+// It will return an error if required information is missing.
+func (n *NutanixClientHelper) buildProviderFromNutanixCluster(nutanixCluster *infrav1.NutanixCluster) (envTypes.Provider, error) {
+	prismCentralInfo := nutanixCluster.Spec.PrismCentral
+	if prismCentralInfo == nil {
+		return nil, nil
+	}
+
+	// PrismCentral is set, build a provider and fixup missing information
+	if prismCentralInfo.Address == "" {
+		return nil, ErrPrismAddressNotSet
+	}
+	if prismCentralInfo.Port == 0 {
+		return nil, ErrPrismPortNotSet
+	}
+	credentialRef, err := nutanixCluster.GetCredentialRefForCluster()
+	if err != nil {
+		//nolint:wrapcheck // error is already wrapped
+		return nil, err
+	}
+	// If namespace is empty, use the cluster namespace
+	if credentialRef.Namespace == "" {
+		credentialRef.Namespace = nutanixCluster.Namespace
+	}
+	additionalTrustBundleRef := prismCentralInfo.AdditionalTrustBundle
+	if additionalTrustBundleRef != nil &&
+		additionalTrustBundleRef.Kind == credentials.NutanixTrustBundleKindConfigMap &&
+		additionalTrustBundleRef.Namespace == "" {
+		additionalTrustBundleRef.Namespace = nutanixCluster.Namespace
+	}
+
+	return kubernetesEnv.NewProvider(*prismCentralInfo, n.secretInformer, n.configMapInformer), nil
+}
+
+// buildProviderFromFile will return an envTypes.Provider with info from the provided file.
+// It will return an error if required information is missing.
+func (n *NutanixClientHelper) buildProviderFromFile() (envTypes.Provider, error) {
```

Comment:
Same thing here, pulled it out into its own function to make it more testable, but the logic stayed the same.